### PR TITLE
Do not assume `sqlite3_column_text` is valid UTF-8.

### PR DESCRIPTION
**Pull request by gwenn**
Fix Statement::value_ref

### CODE REVIEW

**Comment by gwenn** [comment]
See #548

File: src/statement.rs [hunk 1]
```diff
@@ -624,8 +624,7 @@ impl Statement<'_> {
 
                 // sqlite3_column_text returns UTF8 data, so our unwrap here should be fine.
```
Comment (thomcc) [inline src/statement.rs]: Stale comment.

**Comment by gwenn** [comment]
> Another approach might be to return the &[u8] and let the caller handle it. This API means that if you have non utf8-text it can't be accessed.

Indeed !

**Comment by thomcc** [comment]
> Indeed !

Concretely, [this code](https://github.com/mozilla/application-services/blob/master/components/places/src/db/db.rs#L273-L277) spends around 10% of its time validating UTF8. if getting a &[u8] out of a ValueRef were possible, we could avoid this. (That said, it's not particularly hot code, in practice IO takes most of the time, but still...)

Using ValueRef (outside of rusqlite's internals) already implies you're worried about perf some anyway.

**Pull request merged**